src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -317,17 +313,7 @@ public char PeekChar(int delta)
             int position = this.Position;
             this.AdvanceChar(delta);
 
-            char ch;
-            if (_offset >= _characterWindowCount
-                && !MoreChars())
-            {
-                ch = InvalidCharacter;
-            }
-            else
-            {
-                // N.B. MoreChars may update the offset.
-                ch = _characterWindow[_offset];
-            }
+            char ch = PeekChar();
```

Comment:
I would be careful with this... `PeekChar(int)` is a very hot method in the compiler IIRC. I'm worried that calling `PeekChar()` isn't going to be inlined, causing significant overhead from the extra calls.